librz/util/syscmd.c

```diff
@@ -131,37 +131,24 @@ RZ_API RZ_OWN char *rz_syscmd_ls(RZ_NONNULL const char *input) {
 	char *name;
 	char *dir;
 	int off;
-	if (!input) {
-		input = "";
-		path = ".";
-	}
-	if (*input == 'q') {
-		printfmt = 'q';
-		input++;
-	}
-	if (*input && input[0] == ' ') {
-		input++;
-	}
-	if (*input) {
-		if ((!strncmp(input, "-h", 2))) {
+	if (argc > 1) {
+		if ((!strncmp(argv[1], "-h", 2))) {
 			eprintf("Usage: ls ([-e,-l,-j,-q]) ([path]) # long, json, quiet\n");
-		} else if ((!strncmp(input, "-e", 2))) {
+			return NULL;
+		} else if ((!strncmp(argv[1], "-e", 2))) {
 			printfmt = 'e';
-			path = rz_str_trim_head_ro(input + 2);
-		} else if ((!strncmp(input, "-q", 2))) {
+		} else if ((!strncmp(argv[1], "-q", 2))) {
 			printfmt = 'q';
-			path = rz_str_trim_head_ro(input + 2);
-		} else if ((!strncmp(input, "-l", 2)) || (!strncmp(input, "-j", 2))) {
-			printfmt = (input[1] == 'j') ? FMT_JSON : FMT_RAW;
-			path = rz_str_trim_head_ro(input + 2);
-			if (!*path) {
-				path = ".";
-			}
+		} else if ((!strncmp(argv[1], "-l", 2)) || (!strncmp(argv[1], "-j", 2))) {
+			printfmt = (argv[1][1] == 'j') ? FMT_JSON : FMT_RAW;
 		} else {
-			path = input;
+			path = rz_str_trim_head_ro(argv[1]);
```

Comment:
No need for trimming the string

Reply:
Expecting some user to type: ls dir_name directly, where the directory argument is the first argument after 'ls'

Reply:
There is no need to trim - rzshell splits everything as required.